src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -522,10 +523,16 @@ object CheckpointAcceptanceService {
           case (hash, txs) =>
             txChainService
               .getLastAcceptedTransactionRef(hash)
-              .map(txs.headOption.map(_.lastTxRef).contains)
+              .map {
+                latr =>
+                  val prevTxHashes = txs.map(_.lastTxRef.prevHash)
+                  val txHashes = txs.map(_.hash)
+                  (latr.prevHash :: txHashes zip prevTxHashes)
```

Comment:
I wonder if checking hashes only is enough I remember that I also asserted on the ordinals while pulling transactions for consensus.

Reply:
:+1: IMO ordinals should be checked as well as it must be in order with no gaps

Reply:
That's what I think also. So I think we should adjust it.

Reply:
Ok, I will add ordinals checking as well.